log4j-redis/src/main/java/org/apache/logging/log4j/redis/appender/RedisAppender.java

```diff
@@ -0,0 +1,244 @@
+            return sslConfiguration;
+        }
+
+        LoggingJedisPoolConfiguration getPoolConfiguration() {
+            return poolConfiguration;
+        }
+
+        int getPort() {
+            return port;
+        }
+
+        public B setKeys(final String keys) {
+            this.keys = keys;
+            return asBuilder();
+        }
+
+        public B setHost(final String host) {
+            this.host = host;
+            return asBuilder();
+        }
+
+        public B setPort(final int port) {
+            this.port = port;
+            return asBuilder();
+        }
+
+        public B setQueueCapacity(final int queueCapacity) {
+            this.queueCapacity = queueCapacity;
+            return asBuilder();
+        }
+
+        public B setPoolConfiguration(final LoggingJedisPoolConfiguration poolConfiguration) {
+            this.poolConfiguration = poolConfiguration;
+            return asBuilder();
+        }
+
+        public B setSslConfiguration(final SslConfiguration ssl) {
+            this.sslConfiguration = ssl;
+            return asBuilder();
+        }
+
+        public B setImmediateFlush(final boolean immediateFlush) {
+            this.immediateFlush = immediateFlush;
+            return asBuilder();
+        }
+
+        RedisManager getRedisManager() {
+            return new RedisManager(
+                    getConfiguration().getLoggerContext(),
+                    getName(),
+                    getKeys().split(KEY_SEPARATOR),
+                    getHost(),
+                    getPort(),
+                    getSslConfiguration(),
+                    getPoolConfiguration()
+            );
+        }
+    }
+
+    /**
+     * Creates a builder for a RedisAppender.
+     * @return a builder for a RedisAppender.
+     */
+    @PluginBuilderFactory
+    public static <B extends Builder<B>> B newBuilder() {
+        return new Builder<B>().asBuilder();
+    }
+
+    @Override
+    public void append(final LogEvent event) {
+        final Layout<? extends Serializable> layout = getLayout();
+        if (event.getLoggerName() != null && AbstractLogger.getRecursionDepth() > 1) {
+            LOGGER.warn("Recursive logging from [{}] for appender [{}].", event.getLoggerName(), getName());
+        } else if (layout instanceof StringLayout) {
+            logQueue.add(((StringLayout)layout).toSerializable(event));
+            if (shouldFlushLogQueue(event.isEndOfBatch())) {
+                try {
+                    tryFlushQueue();
+                } catch (final Exception e) {
+                    error("Unable to write to Redis in appender [" + getName() + "]", event, e);
+                }
+            }
+        } else {
+            throw new AppenderLoggingException("The Redis appender only supports StringLayouts.");
+        }
+    }
+
+    private boolean shouldFlushLogQueue(boolean endOfBatch) {
+        return immediateFlush || endOfBatch || logQueue.size() >= queueCapacity;
```

Comment:
I think we always want to use the result of `BlockingQueue.offer` instead of `logQueue.size() >= queueCapacity`

Reply:
+1 that's much cleaner